### PR TITLE
Updated README according to new on-chain architecture and flow

### PR DESCRIPTION
**Pull request by swagftw**
This PR updates the README to cover the major upgrades and overhaul in project components and changes in architecture according to the latest update on the decentralized protocol state (more info [here](https://github.com/PowerLoom/pooler/pull/25)), as available in the branch [feat/migtate-to-onchain-arch](https://github.com/PowerLoom/audit-protocol/tree/feat/migrate-to-onchain-arch).

### Checklist
- [x] My branch is up-to-date with upstream/main branch.
- [x] Everything works and tested for major version of Python/NodeJS/Go and above.
- [x] I ran pre-commit checks against my changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,104 +3,83 @@
...
 - [Monitoring and Debugging](#monitoring-and-debugging)
+- [Setting up Slack workflow](#setting-up-slack-workflow)
 - [Architecture Details](#architecture-details)
 
 ## Overview
 
-![Audit-protocol workflow](docs/images/AuditProtocolOverview.png)
-Audit-Protocol is the component of a fully functional, distributed system that works alongside Pooler, off-chain consensus and together they are responsible for
+Architecture overview
+![Architecture](./docs/images/onchain_architecture.png)
```
Comment (xadahiya) [inline README.md:16]: Let's use the latest image from https://github.com/PowerLoom/pooler/pull/25

File: README.md [hunk 2]
```diff
@@ -3,104 +3,83 @@
...
+
+Audit-Protocol is the component of a fully functional, distributed system that works alongside Pooler and Epoch Generator together they are all responsible for
 * generating a time series database of changes occurring over smart contract state data and event logs, that live on decentralized storage protocols
 * higher order aggregated information calculated over decentralized indexes maintained atop the database mentioned above
 
-Audit-protocol can also be used for storing stream of json encoded data from any on-chain source(like a smart contract on a blockchain).
+Audit-protocol is used for committing snapshots to on-chain source (like a smart contract on a blockchain) via transaction relayer or directly to it.
```
Comment (xadahiya) [inline README.md:22]: This line is not very clear, maybe we should make it more specific and elaborate a bit more?

File: README.md [hunk 3]
```diff
@@ -3,104 +3,83 @@
...
 It has the following functions:
 
-1. [Store snapshots](https://www.notion.so/powerloom/PowerLoom-Protocol-Overview-c3bf9dd9323541118d46a4d8684565d1?pvs=4#68d50c3a7db14f2893706e805f083bec) on decentralized storage protocols like IPFS and Filecoin.
-2. Commit [anchor-proof](https://www.notion.so/powerloom/PowerLoom-Protocol-Overview-c3bf9dd9323541118d46a4d8684565d1?pvs=4#56c5333fa16940e8b0437e4d03129700) of snapshot on-chain
-2. Submit snapshots for [consensus](https://github.com/PowerLoom/product/issues/10)
-3. Track the [snapshot finalization](https://www.notion.so/powerloom/PowerLoom-Protocol-Overview-c3bf9dd9323541118d46a4d8684565d1?pvs=4#bf7a29dfd1384d28b45369b98d74a40f) and build a continuous DAG chain of snapshots
-4. Provide API to access the [DAG chain](https://www.notion.so/powerloom/PowerLoom-Protocol-Overview-c3bf9dd9323541118d46a4d8684565d1?pvs=4#c008da3137294ad6b065cfe0fcc05032). of snapshots
-5. Perform various high order operations such as time based indexing and aggregations on top of generated DAG chains
+<!-- TODO: add links to while paper for each of the functions below -->
```
Comment (xadahiya) [inline README.md:25]: while paper? 
Also, if README is not done yet please convert to draft.

File: README.md [hunk 4]
```diff
@@ -3,104 +3,83 @@
...
+1. Store snapshots on decentralized storage protocols like IPFS and Filecoin.
```
Comment (xadahiya) [inline README.md:26]: Are we storing to Filecoin using payload commit service atm?

File: README.md [hunk 5]
```diff
@@ -3,104 +3,83 @@
...
+2. Submit snapshots to the on-chain source for consensus.
+3. Generate snapshotter report for snapshot submissions.
+4. Keep a local cache of finalized snapshots.
+5. Prune older snapshots.
```
Comment (xadahiya) [inline README.md:30]: What exactly does `Prune older snapshots` mean?
I think we need to improve all 5 points and explain things in a better way. README should be easy to understand even for people who are not active contributors.

File: README.md [hunk 6]
```diff
@@ -3,104 +3,83 @@
...
-pm2 start pm2.config.js
+cp settings.example.json settings.json
 ```
 
-### Setup Virtual Env
-* Setup a python virtual environment [Ref](https://docs.python.org/3/library/venv.html)
+- Configure the snapshotter account address (EVM compatible 0x...) in `settings.instance_id`
+- Configure the `pooler-namespace`
+- Configure `anchor_chain_rpc_url` in `settings.anchor_chain_rpc_url`
+- Configure the local cache directory path in `settings.local_cache_path`
```
Comment (xadahiya) [inline README.md:55]: People don't need to configure this as this is fixed atm to `home/ubuntu/local_cache`

File: README.md [hunk 7]
```diff
@@ -3,104 +3,83 @@
...
 
 ```shell
-pm2 start pm2.config.js
+cp settings.example.json settings.json
 ```
 
-### Setup Virtual Env
-* Setup a python virtual environment [Ref](https://docs.python.org/3/library/venv.html)
+- Configure the snapshotter account address (EVM compatible 0x...) in `settings.instance_id`
+- Configure the `pooler-namespace`
```
Comment (xadahiya) [inline README.md:53]: You should explain what exactly is `pooler-namespace`. Take hints from new Pooler README file (https://github.com/PowerLoom/pooler/blob/dockerify/README.md).

**Review by xadahiya — CHANGES_REQUESTED** [review]
Some suggestions to make README easier to understand. 

P.S. - Let's also update architecture diagram in Introduction.md from new Pooler README.

File: README.md (hunk 2 above)
Comment (swagftw) [reply README.md:22]: Improved by adding detailed behaviour

File: README.md (hunk 4 above)
Comment (swagftw) [reply README.md:26]: No, removed `Filecoin` and added `Web3 Storage`

File: README.md (hunk 5 above)
Comment (swagftw) [reply README.md:30]: Improved the  with better explanation

File: README.md (hunk 7 above)
Comment (swagftw) [reply README.md:53]: Updated README

File: README.md [hunk 8]
```diff
@@ -3,104 +3,83 @@
...
+- Configure `anchor_chain_rpc_url` in `settings.anchor_chain_rpc_url`
+- Configure the local cache directory path in `settings.local_cache_path`
+- If you are using 3rd party IPFS provider, add a valid URL in `settings.ipfs.url` and `settings.ipfs.reader_url`
+  - Fill authentication keys for the provider in `settings.ipfs.writer_auth_config` and `settings.ipfs.reader_auth_config`
+- Configure access token for Web3 Storage in `settings.web3_storage.api_token`
```
[the remark this answers is not in the document]
Comment (swagftw) [reply README.md]: moved to optional section

**Review by xadahiya — APPROVED** [review]

**Comment by xadahiya** [comment]
Improvements LGTM! All concerns were addressed.

**Review by anomit — APPROVED** [review]
LGTM

**Pull request merged**